spring-cloud-skipper-shell-commands/src/main/java/org/springframework/cloud/skipper/shell/config/ShellConfiguration.java

```diff
@@ -0,0 +1,116 @@
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.cloud.skipper.client.SkipperClientConfiguration;
+import org.springframework.cloud.skipper.shell.command.support.TargetHolder;
+import org.springframework.cloud.skipper.shell.support.ShellCommandLineParser;
+import org.springframework.cloud.skipper.shell.support.ShellCommandLineRunner;
+import org.springframework.cloud.skipper.shell.support.ShellProperties;
+import org.springframework.context.annotation.*;
+import org.springframework.shell.CommandLine;
+import org.springframework.shell.core.JLineShell;
+import org.springframework.shell.core.JLineShellComponent;
+
+/**
+ * Configures the various commands that are part of the default Spring Shell experience.
+ *
+ * @author Josh Long
+ * @author Mark Pollack
+ * @author Eric Bottard
+ */
+@Configuration
+@ImportResource("classpath*:/META-INF/spring/spring-shell-plugin.xml")
+@Import(SkipperClientConfiguration.class)
+public class ShellConfiguration {
+
+	private static final Logger logger = LoggerFactory.getLogger(ShellConfiguration.class);
+
+	@Autowired
+	private CommandLine commandLine;
+
+	@Bean
+	public TargetHolder targetHolder() {
+		return new TargetHolder();
+	}
+
+	@Bean
+	public ShellCommandLineParser shellCommandLineParser() {
+		return new ShellCommandLineParser();
+	}
+
+	@Bean
+	public ShellProperties shellProperties() {
+		return new ShellProperties();
+	}
+
+	/**
+	 * Return the interactive command line runner. Note: Add a
+	 * {@link ConditionalOnMissingBean} annotation is used so that this interactive command
+	 * line runner is not created when running the shell in the same process as the Gilligan
+	 * server.
+	 *
+	 * @return the interactive shell
+	 */
+	@Bean
+	public ShellCommandLineRunner commandLineRunner() {
+		return new ShellCommandLineRunner();
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(CommandLine.class)
+	public CommandLine commandLine(ShellCommandLineParser shellCommandLineParser, ShellProperties shellProperties,
+			ApplicationArguments applicationArguments) throws Exception {
+		return shellCommandLineParser.parse(shellProperties, applicationArguments.getSourceArgs());
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(JLineShell.class)
+	public JLineShellComponent shell() {
+		return new JLineShellComponent();
+	}
+
+	@Configuration
+	@ComponentScan({ "org.springframework.shell.converters", "org.springframework.shell.plugin.support" })
+	public static class DefaultShellComponents {
+
+		@PostConstruct
+		public void log() {
+			logger.debug(
+					"default (o.s.shell.{converters,plugin.support})" + " Spring Shell packages are being scanned");
+		}
+	}
+
+	@Configuration
+	@ComponentScan({ "org.springframework.shell.commands", "org.springframework.cloud.skipper.shell.command",
+			"org.springframework.cloud.skipper.shell.config" })
```

Comment:
which package?  We want the core shell commands, obviously the skipper ones.  Do you mean  `org.springframework.cloud.skipper.shell.config`?  Those might get picked up by boot package scanning anyway....Can test by removing it and see if the banner and prompt still work as expected.